docs/wiki/transactions/transaction-body.md

```diff
@@ -0,0 +1,85 @@
+| Fee               | Transaction fee                                                                                   | 8 bytes             |
+| Excess commitment | Pedersen commitment                                                                               | 33 bytes            |
+| Excess signature  | Schnorr signature                                                                                 | 64 bytes            |
+
+### Height locked kernels
+
+The first protocol **kernel** would always consist of **Feature**, **Fee**, **Lock height** and standard **Excess commitment** and **Signature**. Starting from the V2 protocol a specific type of kernel - **height locked** kernel supports the **lock height**.
+
+| Field             | Description                                                                                       | Size                |
+|:------------------|:--------------------------------------------------------------------------------------------------|:--------------------|
+| Feature           | **Plain** (0x00), **Coinbase** (0x01), **Height Locked** (0x02) or **No recent duplicate** (0x03) | 1 byte              |
+| Fee               | Transaction fee                                                                                   | 8 bytes             |
+| Lock height       | Block height starting from which the kernel becomes valid                                         | 8 bytes             |
+| Excess commitment | Pedersen commitment                                                                               | 33 bytes            |
+| Excess signature  | Schnorr signature                                                                                 | 64 bytes            |
+
+### NRD (No Recent Duplicate) kernels
+
+When transaction is representing a creation of a payment channel it is relevant to ensure that one of the parties cannot update own kernel associated to own output in the transaction. The NRD kernels are created simultaneously and correlated preventing such an attack. More details in the [mailing list archive](https://lists.launchpad.net/mimblewimble/msg00636.html).
+
+| Field             | Description                                                                                       | Size                |
+|:------------------|:--------------------------------------------------------------------------------------------------|:--------------------|
+| Feature           | **Plain** (0x00), **Coinbase** (0x01), **Height Locked** (0x02) or **No recent duplicate** (0x03) | 1 byte              |
+| Fee               | Transaction fee                                                                                   | 8 bytes             |
+| Lock height       | Block height starting from which the kernel becomes valid                                         | 8 bytes             |
```

Comment:
This is relative lock height:
Number of most recent blocks in which the same NRD must be absent.